### DISCUSSION TITLE
Can't connect to linux server based on amazon linux 2023

### DISCUSSION THREAD

**Discussion by Bishop0948** [Eugeny/tabby#8399] [Q&A]
It's werid that i using lastest tabby to connect to a EC2 instance based on Amazon linux 2023, got this:
![image](https://github.com/Eugeny/tabby/assets/84124160/5a1c5fb9-4629-4561-884a-acfa30f7d0f2)
It always auth twoice, and got error "All configured authentication methods failed". But in powershell it's connectable. (i've lost the pic, but u know what i meant :< )

So i try to use amazon linux 2, it's connected! And only auth once!
![image](https://github.com/Eugeny/tabby/assets/84124160/14d92f43-4698-422d-b467-e7774c2fa4d5)

Why?????

**Comment by tinywolf3** [comment]
same problem.
ubuntu terminal is ok.

**Reply by GordonWang** [reply]
It is Ok from terminal if using `ssh ec2-user@my-aws-ami-2023-ec2-instance`

Looks like it is easy to reproduce. I am using MacOS "Tabby Version 1.0.197 (1.0.197)"

**Comment by syounger** [comment]
This is related to changes Amazon made deprecating rsa authentication on the new 2023 images. You need to generate a new key with a different algorithm.
https://docs.aws.amazon.com/linux/al2023/ug/ssh-host-keys-disabled.html

I don't know why the login still works from a command line but not from Tabby.

**Reply by GordonWang** [reply]
@syounger   Thank you for finding this. 

I generate new ssh key with ed25519 algorithm and set the private key in tabby profile configuration, and it works. 
```bash
ssh-keygen -t ed25519
```

According to amazon's doc, I guess the ssh client lib used by Tabby does not support rsa-sha2-256 or rsa-sha2-512 signature. However, the mac native ssh client supports these more secure rsa signature. This might be the root cause. (I do not verify Tabby's code yet, just my guess)

Anyway, generating other algorithm ssh keys looks like a good solution.
